### PR TITLE
[OpPerf] Add Neural network loss ops

### PR DESCRIPTION
**Pull request by ChaiBapchya**
## Description ##
Added Neural Network loss operators to the existing set of opperf benchmarks

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] Added following ops
1. smooth_l1
2. CTCLoss [ctc_loss added as alias]
3. MakeLoss
4. softmax_cross_entropy

## Results ##
```
from benchmark.opperf.nd_operations.nn_loss_operators import run_loss_operators_benchmarks
 run_loss_operators_benchmarks()

INFO:root:Begin Benchmark - CTCLoss
INFO:root:Complete Benchmark - CTCLoss
INFO:root:Begin Benchmark - MakeLoss
INFO:root:Complete Benchmark - MakeLoss
INFO:root:Begin Benchmark - ctc_loss
INFO:root:Complete Benchmark - ctc_loss
INFO:root:Begin Benchmark - smooth_l1
INFO:root:Complete Benchmark - smooth_l1
INFO:root:Begin Benchmark - softmax_cross_entropy
INFO:root:Complete Benchmark - softmax_cross_entropy
```
CPU - https://gist.github.com/ChaiBapchya/fb608c551689d3a4ff48c3be5030a2e2

## Comments ##
@sandeep-krishnamurthy @apeforest @connorgoggins

### CODE REVIEW

File: benchmark/opperf/nd_operations/nn_loss_operators.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    runs: int, default 100
+        Number of runs to capture benchmark results
+
+    Returns
+    -------
+    Dictionary of results. Key -> Name of the operator, Value -> Benchmark results.
+
+    """
+    # Fetch all loss operators
+    mx_rearrange_ops = get_all_loss_operators()
```
Comment (connorgoggins) [inline benchmark/opperf/nd_operations/nn_loss_operators.py]: nit: shouldn't this variable be called `mx_loss_ops`?

File: benchmark/opperf/nd_operations/nn_loss_operators.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+    # Run benchmarks
+    mx_rearrange_op_results = run_op_benchmarks(mx_rearrange_ops, dtype, ctx, profiler, warmup, runs)
```
Comment (connorgoggins) [inline benchmark/opperf/nd_operations/nn_loss_operators.py]: Same here - `mx_loss_op_results`

File: benchmark/opperf/utils/benchmark_utils.py [hunk 3]
```diff
@@ -27,6 +27,8 @@
 from .profiler_utils import cpp_profile,python_profile
 
 
+no_backward = ['softmax_cross_entropy']
```
Comment (connorgoggins) [inline benchmark/opperf/utils/benchmark_utils.py]: +1, good idea to create this list

**Comment by ChaiBapchya** [comment]
GPU : https://gist.github.com/ChaiBapchya/fac7310f7d167d1361854451e7daa342

**Comment by ChaiBapchya** [comment]
@mxnet-label-bot add [pr-awaiting-review]

File: benchmark/opperf/utils/profiler_utils.py [hunk 4]
```diff
@@ -48,8 +48,8 @@ def _get_operator_profile(operator_name, operator_profile_results):
     # alias map : dictionary of the form {"alias" : "registered_name"}
     # allows to retrieve alias operator profile from the profiler results
     # TODO handling - "identity" : "_copy"
-    alias_map = {"broadcast_plus" : "broadcast_add", "broadcast_minus" : "broadcast_sub", "flatten" : "Flatten", "max_axis" : "max",
-                 "swapaxes" : "SwapAxis", "flip" : "reverse", "reshape" : "Reshape", "crop" : "slice", "sum_axis" : "sum", "min_axis" : "min"}
+    alias_map = {"broadcast_plus": "broadcast_add", "broadcast_minus": "broadcast_sub", "flatten": "Flatten", "max_axis": "max",
```
Comment (apeforest) [inline benchmark/opperf/utils/profiler_utils.py]: nit: I think pep8 requires no space around ':'?

Comment (ChaiBapchya) [reply benchmark/opperf/utils/profiler_utils.py]: pep8 requires no space before ":"
http://pep8online.com/checkresult
https://www.python.org/dev/peps/pep-0008/

**Comment by ChaiBapchya** [comment]
@mxnet-label-bot add [pr-awaiting-merge]
